src/SteppingAction.cpp

```diff
@@ -27,5 +27,10 @@ void SteppingAction::UserSteppingAction(const G4Step* step)
 {
     AnalysisManager * analysis_manager = AnalysisManager::Instance();
 
-    analysis_manager->AddProcess(step->GetPostStepPoint()->GetProcessDefinedStep()->GetProcessName());
+    if (step->GetPostStepPoint()->GetProcessDefinedStep() != 0){
+      event.AddProcess(step->GetPostStepPoint()->GetProcessDefinedStep()->GetProcessName());
+    } else {
+      event.AddProcess("User Limit");
+    }
+
```

Comment:
Why not get the process pointer into a local variable, reducing the copy-pasted code:
```
G4VProcess* stepProc = step->GetPostStepPoint()->GetProcessDefinedStep();
G4String procName = (stepProc ? stepProc->GetProcessName() : "User Limit");
analysis_manager->AddProcess(procName);
```